src/components/AppConfig/AppConfig.tsx

```diff
@@ -86,18 +92,24 @@ export const AppConfig = ({ plugin }: AppConfigProps) => {
                 secureJsonData[key] = newSecrets[key];
               }
             }
-            updatePluginAndReload(plugin.meta.id, {
+            await updatePlugin(plugin.meta.id, {
               enabled,
               pinned,
               jsonData: settings,
               secureJsonData,
-            })
+            });
+            const result = await checkPluginHealth(plugin.meta.id);
+            setIsUpdating(false);
+            setHealthCheck(result.data);
           }}
           disabled={!updated}
         >
-          Save settings
+          Save &amp; test
         </Button>
+        {isUpdating && <Spinner />}
```

Comment:
Instead of rendering a spinner out of the button, you can use [LoadingPlaceholder](https://developers.grafana.com/ui/latest/index.html?path=/story/general-loadingplaceholder--basic) to replace the button content while loading.

Reply:
I've done something similar in the latest commit - take a look and see what you think?

Reply:
There might be existing UI patterns that I'm not aware of here but I haven't noticed buttons disappearing while things are loading 🤔